### PR TITLE
Markdown: Fix Markdown converting '<' to &lt; inside backticks

### PR DESCRIPTION
**Pull request by jllano**
- markdown is unable to process '<' signs within the code block tag (backticks)
- markdown convert '<' signs into '&amp;amp;lt;' after the text proccessing process
- therefore we need restore the entity back into its original name '&amp;lt;'

Fixes #22175

### CODE REVIEW

**Comment by jllano** [comment]
> to ensure the text processing (CoreFormatting) remain in possession in terms of security and processing

my apologies for the confusion, I will removed that line.

> preg_replace( '#&amp;lt;.+?.*?&gt;#i', '$0', $t_string )

Good catch, no need to add a preg_replace also, thank you

**Comment by jllano** [comment]
Before:
screen shot 2017-01-19 at 8 54 55 pm

After:
screen shot 2017-01-19 at 8 55 11 pm

File: plugins/MantisCoreFormatting/MantisCoreFormatting.php [hunk 1]
```diff
@@ -131,9 +131,10 @@ function ( ) {
 
 		$t_string = MantisMarkdown::convert_text( $t_string );
 
-		# restore '<' converting &lt; within backticks 
-		$t_string = str_replace( '&amp;lt;', '&lt;', preg_replace( '#&amp;lt;.+?.*?&gt;#i', '$0', $t_string ));
-			
+		# restore '<' signs converting &lt; within code block tag (backticks)
+		# markdown coudnt perform well once the text procces executed the text
```
Comment (dregad) [inline plugins/MantisCoreFormatting/MantisCoreFormatting.php]: 2nd line of comment does not make sense. 
- coundt -> doesn'nt ?
- define what _perform well_ means
- _the text procces_ (processing ?) _executed the text_ what does that mean ?

**Comment by atrol** [comment]
There is 

> Fixes #22175,#22204

in the first commit message.
I don't see at the moment how this change fixes http://www.mantisbt.org/bugs/view.php?id=22204

**Comment by jllano** [comment]
It does @atrol, please checkout screenshots above.

**Comment by atrol** [comment]
Are we talking about the same thing?

I am talking about main_page.php with $g_news_enabled = ON;
@badfiles, is this what you mean in http://www.mantisbt.org/bugs/view.php?id=22204 ?

**Comment by jllano** [comment]
I assumed @badfiles is referring the <h1> to <h6> tags are used to define HTML headings?

**Comment by dregad** [comment]
@jllano see http://www.mantisbt.org/bugs/view.php?id=22204#c55241 - definitely **not** fixed by this PR (I just tested). Please update commit message to remove bug ID reference.

**Comment by jllano** [comment]
@dregad I updated the commit, I think its much cleaner now.

Thanks

**Comment by vboctor** [comment]
I've merged the fix for the issue, but will not merge the unit tests for now.  Will open a separate work item for revising how unit tests for markdown should be implemented so that:

- We are testing the production code, and not a copy of it.
- Making sure that markdown tests run when MantisBT tests run as part of Travis.

**Pull request closed without merging**